lib/hanami/cli/generators/gem/app.rb

```diff
@@ -68,6 +68,15 @@ def generate_app(app, context) # rubocop:disable Metrics/AbcSize
               fs.write("app/assets/images/favicon.ico", file("favicon.ico"))
             end
 
+            if context.generate_db?
+              fs.write("app/repo.rb", t("repo.erb", context))
```

Comment:
Yea, that makes sense to me. Should we also nest in a DB namespace like the other files in that folder? I feel like it should for consistency's sake. It could go either way since Repos are the bridge between app logic and persistence, so I don't feel strongly either way.

Reply:
Thanks for the extra nudge about this question, @cllns!

Here's a decision for you: let's generate this into `db/repo.rb`.

With this change, we:

- Have the namespace of the class within the app echo the namespace of its superclass
- Put the app's repo base class alongside the struct base class, which feels sensible since its the job of repos to return struct instances in the first place
- Keep all the db-related support code together in one place
- Reinforce the location of `db/` as the home for all db-support code, whether this is extra ROM components (like changesets or mappers), or arbitrary code that the user wants to mix in or use alongside the other db components
- Subtly reinforce that a user may choose to have _other_ kinds of repo base classes, if they e.g. want to mix and match ROM-based repos with similarly-behaving repos of their own construction

Feels like you were happy to move in this direction, but I'm also happy to discuss further if you have any other thoughts or questions 😄